backend/program/scrapers/jackett.py

```diff
@@ -67,29 +67,30 @@ def _scrape_item(self, item):
 
     def api_scrape(self, item):
-        """Wrapper for torrentio scrape method"""
-        query = ""
-        if item.type == "movie":
-            query = f"&t=movie&imdbid={item.imdb_id}"
-        if item.type == "season":
-            query = f"&t=tv-search&imdbid={item.parent.imdb_id}&season={item.number}"
-        if item.type == "episode":
-            query = f"&t=tv-search&imdbid={item.parent.parent.imdb_id}&season={item.parent.number}&ep={item.number}"
-
-        url = (
-            f"{self.settings.url}/api/v2.0/indexers/all/results/torznab?apikey={self.api_key}{query}"
-        )
-        with self.second_limiter:
-            response = get(url=url, retry_if_failed=False, timeout=60)
-        if response.is_ok:
-            data = {}
-            for stream in response.data['rss']['channel']['item']:
-                title = stream.get('title')
-                for attr in stream.get('torznab:attr', []):
-                    if attr.get('@name') == 'infohash':
-                        infohash = attr.get('@value')
-                if parser.parse(title) and infohash:
-                    data[infohash] = {"name": title}
-            if len(data) > 0:
-                return parser.sort_streams(data)
-        return {}
+        """Wrapper for `Jackett` scrape method"""
+        # https://github.com/Jackett/Jackett/wiki/Jackett-Categories
+        with self.minute_limiter:
+            query = ""
+            if item.type == "movie":
+                query = f"&cat=2010,2040,2045,2050,2080&t=movie&q={item.title}&year={item.aired_at.year}"
+            if item.type == "season":
+                query = f"&cat=5010,5020,5040,5045,5060,5070,5080&t=tvsearch&q={item.parent.title}&season={item.number}"
+            if item.type == "episode":
+                query = f"&cat=5010,5020,5040,5045,5060,5070,5080&t=tvsearch&q={item.parent.parent.title}&season={item.parent.number}&ep={item.number}"
+            url = (f"{self.settings.url}/api/v2.0/indexers/!status:failing,test:passed/results/torznab?apikey={self.api_key}{query}")
+            with self.second_limiter:
+                response = get(url=url, retry_if_failed=False, timeout=60)
+            if response.is_ok:
+                data = {}
+                for stream in response.data['rss']['channel'].get('item', []):
+                    title = stream.get('title')
+                    if parser.check_for_title_match(item, title):
+                        if parser.parse(title):
+                            attr = stream.get('torznab:attr', [])
+                            infohash_attr = next((a for a in attr if a.get('@name') == 'infohash'), None)
+                            if infohash_attr:
+                                infohash = infohash_attr.get('@value')
+                                data[infohash] = {"name": title}
+                if len(data) > 0:
+                    return parser.sort_streams(data)

```

Comment:
This could be moved to parser __init__ to avoid duplication in parsers

Reply:
Agreed, ultimately I'd like to aggregate to the parent class and sort there instead of base classes